packages/bot-web-ui/src/stores/save-modal-store.ts

```diff
@@ -75,9 +77,58 @@ export default class SaveModalStore implements ISaveModalStore {
 
             this.setButtonStatus(button_status.COMPLETED);
         }
-
-        this.updateBotName(bot_name);
-        saveWorkspaceToRecent(xml, is_local ? save_types.LOCAL : save_types.GOOGLE_DRIVE);
+        const { active_tab } = this.root_store.dashboard;
+        const { selected_strategy_id, setRecentStrategies } = this.root_store.load_modal;
+
+        if (active_tab === 0) {
+            const workspace_id = selected_strategy_id || Blockly.utils.genUid();
+            const workspace = await getSavedWorkspaces();
+            const current_workspace_index = workspace.findIndex(strategy => strategy.id === workspace_id);
+            let save_type = save_types.UNSAVED;
+
+            if (is_local) {
+                save_type = save_types.LOCAL;
+            } else {
+                save_type = save_types.GOOGLE_DRIVE;
+            }
+            if (save_as_collection) {
+                save_type = save_types.UNSAVED;
+            }
+
+            if (current_workspace_index >= 0) {
+                const current_workspace = {};
+                current_workspace.id = workspace_id;
+                current_workspace.xml = Blockly.Xml.domToText(xml);
+                current_workspace.name = bot_name;
+                current_workspace.timestamp = Date.now();
+                current_workspace.save_type = save_type;
+                workspace[current_workspace_index] = current_workspace;
+            } else {
+                workspace.push({
+                    id: workspace_id,
+                    timestamp: Date.now(),
+                    name: bot_name,
+                    xml: Blockly.Xml.domToText(xml),
+                    save_type,
+                });
```

Comment:
@rupato-deriv here object has the same structure with the same values as `current_workspace`, so we can avoid duplication 
```suggestion
                workspace.push(current_workspace);
```

Reply:
here we cannot use the current object `current_workspace` because its not in the scope of the else check